### PR TITLE
PD-16 Parser development

### PR DESCRIPTION
**Pull request by crapper**
Add a package dependency lodash.isEqual

### CODE REVIEW

**Comment by Jerrylum** [comment]
Why do you need to add `lodash.isEqual` dependency?

**Comment by crapper** [comment]
> Why do you need to add `lodash.isEqual` dependency?

Found that the Implemented token classes e.g. StringT, has no implementation to handle the deep equal operation.
And the Parameter class needs to implement a deep equal function.

File: src/token/Tokens.ts [hunk 1]
```diff
@@ -563,51 +562,135 @@ export class Zero extends Token {
...
+      return false;
+    }
+
+    if (!(obj instanceof Parameter)) {
+      return false;
+    }
 
+    let other = obj as Parameter;
 
+    return other.string == this.string && other.number == this.number && other.bool == this.bool;
```
Comment (Jerrylum) [inline src/token/Tokens.ts:649]: Oh, I see you are working on the equals function in the Parameter class

**Review by Jerrylum — COMMENTED** [review]
That's maybe the reason why you are trying to implement the equals function in the abstract Token class, right? for every sub-classes that extends Token class

**Review by Jerrylum — APPROVED** [review]
I see the Parameter class and the test cases are unfinished, but I can approve this pull request, keep working!

**Pull request merged**